### PR TITLE
numbat: 1.6.3 -> 1.7.0

### PR DESCRIPTION
**Pull request by giomf**
## Description of changes
https://github.com/sharkdp/numbat/releases/tag/v1.7.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by giomf** [comment]
Result of `nixpkgs-review pr 265345` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>numbat</li>
  </ul>
</details>

File: pkgs/by-name/nu/numbat/package.nix [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
   };
 
-  cargoHash = "sha256-MPqJjCfIwgK8QigWQYfWAYlg9RNMzF4x+0SprS0raKY=";
+  cargoHash = "sha256-skCvIMz50GqveyawVHicpEqMknvZuYNvp1u+gW4wG8A=";
 
   buildInputs = lib.optionals stdenv.isDarwin [
     darwin.apple_sdk.frameworks.Security
   ];
 
+  passthru.tests.version = testers.testVersion {
```
Comment (giomf) [inline pkgs/by-name/nu/numbat/package.nix:27]: Is this how I works?
As I understand it, that should call "numbat --version" and check if the version number appears anywhere on stdout?

Comment (natsukium) [reply pkgs/by-name/nu/numbat/package.nix:27]: That's right, testVersion actually executes `${mainProgram} --version` or something and checks whether the stdout contains the correct version.

https://github.com/NixOS/nixpkgs/blob/38f6bf6c9cad76b8128c914d8818150d4a28e781/pkgs/build-support/testers/default.nix#L56-L78

You can check on local machine with the following command.
```sh
$ nix-build -A numbat.tests

checking outputs of '/nix/store/yf41i4sh2nkhv1j7pmvzlyvbi79klgns-numbat-1.7.0-test-version.drv'...
numbat 1.7.0
/nix/store/r7h5n8ywyxw5jp95wg6jh8xwvzjsg57g-numbat-1.7.0-test-version
```

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**